### ISSUE TITLE
How to set column width?

### ISSUE THREAD

**Issue by YannisMarios** [react-bootstrap-table/react-bootstrap-table2#226]
Hi,

I am trying to set the column width? How do you do It with react-bootstrap-table2?

**Comment by AllenFang** [comment]
@YannisMarios https://github.com/react-bootstrap-table/react-bootstrap-table2/issues/221

This is not implemented yet. I will enhance this feature ASAP, thanks.

**Comment by waynebrantley** [comment]
@AllenFang Trying to convert to your new version.  Ran into this same issue - pretty much blocking us at this point.

**Comment by AllenFang** [comment]
@waynebrantley right now, `headerStyle` is just a way, I'm think other good solution for it. I apologize for this inconvenience

**Comment by Northshoot** [comment]
I think current implementation if great!

Just add to your column description and done. 

```
 headerStyle: (colum, colIndex) => {
          return { width: '80px', textAlign: 'center' };
        }
```

**Comment by waynebrantley** [comment]
@AllenFang so is headerStyle going to be how this is done?  That is fine, might should put a storybook showing?

**Comment by nthgol** [comment]
can anyone suggest a way to 'width - auto' size the columns according to the data?

**Comment by AlessandroGambaro** [comment]
<BootstrapTable....>
 <TableHeaderColumn width={'20%'} ....
<TableHeaderColumn width={'80%'} ...
</BootstrapTable>

**Comment by pceccato** [comment]
Is there an example of how to set the column widths? This is fundamental functionality...

**Comment by waynebrantley** [comment]
@pceccato See @Northshoot comment above...that is part of the column definition...works good.

**Comment by pceccato** [comment]
Does it  need to be an arrow function?

**Comment by B-R-Bender** [comment]
> Does it need to be an arrow function?

No, you could use anonymus function declaration. Or any other you may like.

**Comment by JenniferMarston** [comment]
Hi.  I'm just starting to use the component, but the proposed does not work if we want the table to be responsive.

**Comment by ehtasham89** [comment]
how to set width in react-bootstrap-table2
`let columns = [
{
        dataField: 'lastModified',
        text: 'Last Modified Date', 
        width: "60"
      },
]`

Its not working...

**Comment by JenniferMarston** [comment]
Try adding 60px instead.

That should work, but then the table loses its responsive behavior.

**Comment by sahilvicesoftware** [comment]
can we add any icon in column headers?
in 'react-bootstrap-table-next'

**Comment by lovato** [comment]
Response is above: https://github.com/react-bootstrap-table/react-bootstrap-table2/issues/226#issuecomment-375913180
It works.

**Comment by mhdfaisal** [comment]
> I think current implementation if great!
> 
> Just add to your column description and done.
> 
> ```
>  headerStyle: (colum, colIndex) => {
>           return { width: '80px', textAlign: 'center' };
>         }
> ```
It works

**Comment by Maneeshachandran** [comment]
Hi,
  I am trying to increase the width of some particular columns. Adding style to the column description doesn't help. How to achieve my requirement?

**Comment by Shannibrst** [comment]
Hello Same problem how to set column width
above examples, not works for me

Thanks

**Comment by lovato** [comment]
@Maneeshachandran and @Shannibrst ... can you share the code?

**Comment by Maneeshachandran** [comment]
@lovato I need to increase the width of the first column. So I have used the style. But it is not working. 
 var Columns = [
      {
        dataField: "description",
        text: "Description",
        editable: false,
        style:{
           width: '10%'
         }     
     },
      {
        dataField: "deliverableSatus",
        text: "Deliverable Status",
        editor: {
          type: Type.SELECT,
          options: [
            {
              value: "Complete",
              label: "Complete"
            },
            {
              value: "Incomplete",
              label: "Incomplete"
            },
            {
              value: "N/A",
              label: "N/A"
            }
          ]
        }
}
]

**Comment by lovato** [comment]
@Maneeshachandran have you tried previous responses? That worked for me, and it is different from your approach.
Try:
```
var Columns = [
  {
    dataField: "description",
    text: "Description",
    editable: false,
    headerStyle: () => {
      return { width: "10%" };
    }
  },
  {
    dataField: "deliverableSatus",
    text: "Deliverable Status",
    editor: {
      type: Type.SELECT,
      options: [
        {
          value: "Complete",
          label: "Complete"
        },
        {
          value: "Incomplete",
          label: "Incomplete"
        },
        {
          value: "N/A",
          label: "N/A"
        }
      ]
    }
  }
];

```

**Comment by Maneeshachandran** [comment]
@lovato Hi Lovato.. It is working for a simple table. Whereas in my case, I need to implement it for a large table. I have vertical and horizontal scroll bars for the table. For this table, I cannot able to fix the column width.

**Comment by lovato** [comment]
@Maneeshachandran Sorry, but w/o code for this "large table", I can't understand the problem. I am using it across two projects, different table sizes, several tables, and everything is running smooth.

**Comment by davidsonsns** [comment]
I just followed this reference and it worked for me [react-bootstrap-table2 storybook Work on Header Columns - Customize Column Style](https://react-bootstrap-table.github.io/react-bootstrap-table2/storybook/index.html?selectedKind=Work%20on%20Header%20Columns&selectedStory=Customize%20Column%20Style&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel)

**Comment by almog-kiki** [comment]
.react-bootstrap-table th[data-row-selection] {
  width: 40px !important;
}

**Comment by AlexPuga** [comment]
**css**
`.react-bootstrap-table
    display: block
    width: 100%
    overflow-x: auto
    -webkit-overflow-scrolling: touch

.table th,
.table td
    white-space: nowrap`

**Table Rows Options:**
`headerStyle:{minWidth: '200px'}`

**Comment by ddouglas** [comment]
I have no doubt that all of these solutions work flawlessly, however I am hiding the header row:
```
        {
            headerStyle: () => {
                return { width: "70%" };
            },
            dataField: "name",
            text: "Name",
            headerAttrs: {
                hidden: true
            },
            formatter: (_, row) => {
                return (
                    <div>
                        <Image
                            src={IMAGE_URL + `/alliance/` + row.id + `_32.png`}
                            rounded
                            fluid
                            className="mr-3"
                        ></Image>
                        <LinkContainer to={`/alliances/` + row.id}>
                            <a>{row.name}</a>
                        </LinkContainer>
                    </div>
                );
            }
        },

```

 and due to this none of the proposed solutions are working. I can even see the classes being applied using Chrome Dev Tools. The issue, for me, is the default style applied to the wrapper class:

```
.react-bootstrap-table table {
    table-layout: fixed;
}
```

Using dev tools, if I remove that class, the table width adopts the values of the width parameters passes via to BS Tables. I need away to remove this class via the library. I have tried passing in null and a blank string to no avail.
I also apologize because I am not that savy with CSS and I am just learning React, so please excuse any ignorance. I have also read the documentation and could not find a solution.

Thank You.

**Comment by awitherow** [comment]
is this going to be done (been about a year now) or just hacked?

**Comment by Guiw5** [comment]
@AllenFang @YannisMarios 
I think it should be closed as it's solved by an alternative way to do it.
...as we said "it's not an issue is a feature"

**Comment by ddouglas** [comment]
Add the bootstrap4 property and the header overrides I meant above take effect.

**Comment by Nandiniyadavraj** [comment]
hello there, 
I checked on all possible way to increase column width as mentioned above but none of it work for me. 

              headerStyle: (colum, colIndex) => {
                    return { 'white-space': 'nowrap', width: '100px' };
                }
with the help of this code I'm able to manage table header column size but not table cells size.
Even I applied css also on table ...  help me out if any of you had solved this issue.

Thanks.

**Comment by AlexPuga** [comment]
Try it:
`formatter: (col, row) => {
                return <span style={{display: 'block', width: 10, overflow: 'hidden', whiteSpace: 'nowrap', textOverflow: 'ellipsis'}}>{col}</span>
            },`

**Comment by Nandiniyadavraj** [comment]
It's really working. Thank you so much @AlexPuga

**Comment by ChaitraK2000** [comment]
Hi I am trying React-bootstrap-table. But i wanted to increase the width of header row. How can i do that?

**Comment by yousefshakoury** [comment]
> I think current implementation if great!
> 
> Just add to your column description and done.
> 
> ```
>  headerStyle: (colum, colIndex) => {
>           return { width: '80px', textAlign: 'center' };
>         }
> ```

Thanks

**Comment by Sharmistha24** [comment]
Try this.
                {
			dataField: 'about',
			text: 'About',
			headerStyle: () => {
				return { minWidth: '200px' };
			  }
		}

**Comment by tushar681** [comment]
Hi there,
I am facing issue because I am using a table without header, it is hidden as below:
 headerAttrs: {
                hidden: true
            }
As it is hidden, so my headerStyle is not working on it. 
Is there any way, I can set width of table column without headerStyle and only using style :{} attribute or any other way.

Thanks in advance

**Comment by istiak227** [comment]
> Try it:
> `formatter: (col, row) => { return <span style={{display: 'block', width: 100, overflow: 'hidden', whiteSpace: 'nowrap', textOverflow: 'ellipsis'}}>{col}</span> }`

if nothing works, try this one. Worked for me. Cheers!
